### PR TITLE
[1LP][RFR] Automate test: Create schedule for base report

### PR DESCRIPTION
**Pull request by valaparthvi**
This PR adds a test case, which creates a schedule for a base report and asserts the schedule was created without any error.

{{pytest: cfme/tests/intelligence/reports/test_crud.py::test_reports_crud_schedule_for_base_report_once --use-template-cache -sqvvv}}

### CODE REVIEW

File: cfme/tests/intelligence/reports/test_crud.py [hunk 1]
```diff
@@ -264,3 +264,23 @@ def test_reports_delete_saved_report(appliance, request):
...
+    report = appliance.collections.reports.instantiate(
+        type="Configuration Management",
+        subtype="Virtual Machines",
+        menu_name="Hardware Information for VMs",
+    )
+    view = navigate_to(report, "Details")
+    view.configuration.item_select("Add a new Schedule")
+    add_view = report.create_view(NewScheduleView)
+    data = {"hour": "12", "minute": "10"}
+    add_view.fill(data)
```
[the remark this answers is not in the document]
Comment (valaparthvi) [reply cfme/tests/intelligence/reports/test_crud.py]: When I go to the Details page of a report and click on 'Add new Schedule', it opens up the form with some details pre-filled, I want to preserve those details and only make a few changes, before adding the schedule. Using the `create` method would direct to 'Add new Schedule' page and load an empty form.

[the remark this answers is not in the document]
Comment (valaparthvi) [reply cfme/tests/intelligence/reports/test_crud.py]: And I am only creating a schedule here, not deleting it, so I don't think it's CRUD operation. Although, I could change the test name to something shorter.

[the remark this answers is not in the document]
Comment (digitronik) [reply cfme/tests/intelligence/reports/test_crud.py]: better you can import this `NewScheduleView` in `report.py`. create a method in `Report` class.

**Comment by valaparthvi** [comment]
@digitronik 
I have made the changes you requested.
1) Add a `ReportSchedule` view, which can be used to add a schedule from report's `Details` page.
2) Add a `create_schedule` method, which creates a schedule for the base report.
3) Modify test in accordance with the new changes.

File: cfme/intelligence/reports/reports.py [hunk 2]
```diff
@@ -312,6 +313,42 @@ def delete(self, cancel=False):
     def saved_reports(self):
         return self.collections.saved_reports
 
+    def create_schedule(self, name=None, description=None, active=None,
```
Comment (digitronik) [inline cfme/intelligence/reports/reports.py]: Required:
Why all argument default value `None`?

File: cfme/intelligence/reports/reports.py [hunk 3]
```diff
@@ -312,6 +313,42 @@ def delete(self, cancel=False):
...
+            "send_txt": email_options.get("send_txt"),
+            "send_csv": email_options.get("send_csv"),
+            "send_pdf": email_options.get("send_pdf")
+        })
+
+        view.add_button.click()
+        view.flash.assert_no_error()
+
+        schedule = self.appliance.collections.schedules.instantiate(
+            name=name or self.menu_name,
```
Comment (digitronik) [inline cfme/intelligence/reports/reports.py]: Question:
can you please explain this?

File: cfme/tests/intelligence/reports/test_crud.py [hunk 4]
```diff
@@ -264,3 +263,26 @@ def test_reports_delete_saved_report(appliance, request):
     view.configuration.item_select(
         item='Delete selected Saved Reports', handle_alert=True)
     assert not report.exists
+
+
+def test_reports_create_schedule_for_base_report_once(appliance, request):
```
Comment (digitronik) [inline cfme/tests/intelligence/reports/test_crud.py]: Required:
This should be a crud operation.
Please create and delete this scheduled report properly and assert for both
remove finalizer

File: cfme/intelligence/reports/reports.py [hunk 5]
```diff
@@ -556,6 +593,20 @@ def step(self):
...
 
+@navigator.register(Report, "ScheduleReport")
+class ReportSchedule(CFMENavigateStep):
+    VIEW = NewScheduleView
+    prerequisite = NavigateToAttribute("appliance.server", "CloudIntelReports")
+
+    def step(self):
+        self.prerequisite_view.reports.tree.click_path(
+            "All Reports",
+            self.obj.type or self.obj.company_name,
```
Comment (digitronik) [inline cfme/intelligence/reports/reports.py]: I am suspect about these or. are those required?

File: cfme/intelligence/reports/reports.py (hunk 2 above)
Comment (valaparthvi) [reply cfme/intelligence/reports/reports.py]: That is in the case when no argument value is passed in the data.

File: cfme/intelligence/reports/reports.py (hunk 3 above)
Comment (valaparthvi) [reply cfme/intelligence/reports/reports.py]: This instantiates the newly created schedule and returns it.

File: cfme/intelligence/reports/reports.py (hunk 5 above)
Comment (valaparthvi) [reply cfme/intelligence/reports/reports.py]: It is required to navigate to the details page of the report and then select `Add a new Schedule` from the configuration menu.

**Comment by valaparthvi** [comment]
@digitronik I have added a new `ReportScheduleView`, to fix the failing `is_displayed` method. I had done a few things incorrectly, but have now fixed it.

File: cfme/intelligence/reports/reports.py [hunk 6]
```diff
@@ -225,6 +226,24 @@ def is_displayed(self):
         )
 
 
+class ReportScheduleView(SchedulesFormCommon):
```
Comment (mshriver) [inline cfme/intelligence/reports/reports.py]: Hang on I'm pretty sure all of this already exists in the schedules.py module:
https://github.com/ManageIQ/integration_tests/blob/master/cfme/intelligence/reports/schedules.py

File: cfme/tests/intelligence/reports/test_crud.py [hunk 7]
```diff
@@ -264,3 +263,26 @@ def test_reports_delete_saved_report(appliance, request):
     view.configuration.item_select(
         item='Delete selected Saved Reports', handle_alert=True)
     assert not report.exists
+
+
+def test_reports_crud_schedule_for_base_report_once(appliance, request):
```
Comment (mshriver) [inline cfme/tests/intelligence/reports/test_crud.py]: I think this case is covered by test_schedule_queue?  

https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/intelligence/reports/test_reports_schedules.py

This uses several schedules from the cfme/data/ directory.

**Comment by valaparthvi** [comment]
Hey @mshriver, we have views and objects to create a completely new schedule, but as far as I know, there is no way to create a schedule for an existing report. Take this for example:
1) Navigate to a report's `Details` page.
2) Click on `Configuration` and select `Add a new schedule`.
3) You'll see a form with some pre-filled data such as `Filters`, `Name` and `Description`.

Take a look at [this](https://up1-vrutkovs.int.open.paas.redhat.com/#oyZ7jTaXqjU8xs6ig-Uv_g).
I don't think we have any view or object that can directly create a schedule for a report like this.

And the test case is similar to `test_schedule_queue`, but `test_reports_crud_schedule_for_base_report_once` uses the new view and method.

But, there is one drawback I hadn't noticed before. The method `create_schedule` will fail if no data is passed since the `Add` button on New Schedule page does not activate until and unless some change is made in the form.

Let me know what you think.

**Pull request merged**